### PR TITLE
add float16 mode and bug fix around image stacks

### PR DESCRIPTION
**Pull request by k-haru**
Hi,

I added f16 mode and bug fix around image stacks.
It could read relion_project outputs like projection stacks.

### CODE REVIEW

File: src/header.jl [hunk 1]
```diff
@@ -299,7 +299,7 @@ Statistics.mean(h::MRCHeader) = h.dmean
 
 function mrcdatatype(nz, mz, ispg)
     return if ispg == 0
-        ifelse(mz == 1, Image, ImageStack)
+        ifelse(nz == 1, Image, ImageStack)
```
Comment (sethaxen) [inline src/header.jl:302]: Can you add a test that catches this bug?

File: src/MRCFile.jl [hunk 2]
```diff
@@ -64,6 +64,7 @@ const MODE_TO_TYPE = Dict(
     Int32(3) => Complex{Int16},
     Int32(4) => ComplexF32,
     Int32(6) => UInt16,
+    Int32(12) => Float16
```
Comment (sethaxen) [inline src/MRCFile.jl:67]: Can you add some test data and a test that loads it?

**Review by sethaxen — CHANGES_REQUESTED** [review]
Thanks for the PR. This seems to just be missing tests.

**Comment by k-haru** [comment]
How about this?
It should be OK.

**Comment by sethaxen** [comment]
Hi @k-haru, why was this closed?

**Pull request closed without merging**